test/e2e/space_test.go

```diff
@@ -241,6 +242,142 @@ func TestSpaceRoles(t *testing.T) {
+				nsTmplSet.Spec.SpaceRoles = []toolchainv1alpha1.NSTemplateSetSpaceRole{
+					{
+						TemplateRef: adminTierTmpl.Name,
+						Usernames:   []string{"user1", "user2"},
+					},
+				}
+			})
+
+			// then
+			require.NoError(t, err)
+			_, err = memberAwait.WaitForNSTmplSet(nsTmplSet.Name, UntilNSTemplateSetHasConditions(Provisioned()))
+			require.NoError(t, err)
+			// fetch the namespace check the `last-applied-space-roles` annotation
+			ns, err := memberAwait.WaitForNamespace(s.Name, nsTmplSet.Spec.Namespaces[0].TemplateRef, "appstudio",
+				UntilNamespaceIsActive(),
+				UntilHasLastAppliedSpaceRoles(nsTmplSet.Spec.SpaceRoles))
+			require.NoError(t, err)
+			// check that the `admin` role and role bindings were created
+			_, err = memberAwait.WaitForRole(ns, "space-admin")
+			assert.NoError(t, err)
+			_, err = memberAwait.WaitForRoleBinding(ns, "user1-space-admin")
+			assert.NoError(t, err)
+			_, err = memberAwait.WaitForRoleBinding(ns, "user2-space-admin")
+			assert.NoError(t, err)
+
+			t.Run("update templateref", func(t *testing.T) {
+				// given
+				adminTierTmpl2 := NewTierTemplate(t, hostAwait.Namespace, "space-role-admin-2-123456", "space-role-admin-2", "appstudio", "123456", []byte(spaceAdminTmpl2))
+				err := hostAwait.CreateWithCleanup(context.TODO(), adminTierTmpl2)
+				require.NoError(t, err)
+				// when
+				nsTmplSet, err = memberAwait.UpdateNSTemplateSet(nsTmplSet.Name, func(nsTmplSet *toolchainv1alpha1.NSTemplateSet) {
+					nsTmplSet.Spec.SpaceRoles = []toolchainv1alpha1.NSTemplateSetSpaceRole{
+						{
+							TemplateRef: adminTierTmpl2.Name,
+							Usernames:   []string{"user1", "user2"},
+						},
+					}
+				})
+				// then
+				require.NoError(t, err)
+				_, err = memberAwait.WaitForNSTmplSet(nsTmplSet.Name, UntilNSTemplateSetHasConditions(Provisioned()))
+				require.NoError(t, err)
+				// fetch the namespace check the `last-applied-space-roles` annotation
+				ns, err := memberAwait.WaitForNamespace(s.Name, nsTmplSet.Spec.Namespaces[0].TemplateRef, "appstudio",
+					UntilNamespaceIsActive(),
+					UntilHasLastAppliedSpaceRoles(nsTmplSet.Spec.SpaceRoles))
+				require.NoError(t, err)
+				// check that the new `admin 2` role and role bindings were created
+				_, err = memberAwait.WaitForRole(ns, "space-admin-2")
+				assert.NoError(t, err)
+				_, err = memberAwait.WaitForRoleBinding(ns, "user1-space-admin-2")
+				assert.NoError(t, err)
+				_, err = memberAwait.WaitForRoleBinding(ns, "user2-space-admin-2")
+				assert.NoError(t, err)
+				// also, check that the old `admin` role and role bindings were deleted
+				err = memberAwait.WaitUntilRoleDeleted(ns, "space-admin")
+				assert.NoError(t, err)
+				err = memberAwait.WaitUntilRoleBindingDeleted(ns, "user1-space-admin")
+				assert.NoError(t, err)
+				err = memberAwait.WaitUntilRoleBindingDeleted(ns, "user2-space-admin")
+				assert.NoError(t, err)
+
+			})
+		})
+
+		t.Run("with same content", func(t *testing.T) {
```

Comment:
I don't think that we need this test here - it doesn't test anything extra which wouldn't be covered in the previous test. 
But I'll leave it up to you ;-)

Reply:
yeah, probably, but I'd rather keep it for now. We can remove it later if it causes too much maintenance

Reply:
it's not only about the maintenance, but also the execution time :wink:

Reply:
right :/ let me remove it, then